extensions/wikia/AnalyticsEngine/AnalyticsProviderUbisoft.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+class AnalyticsProviderUbisoft implements iAnalyticsProvider {
+
+    const TEMPLATE_PATH = 'extensions/wikia/AnalyticsEngine/templates/ubisoft.mustache';
+    const DISABLED_MESSAGE = '<!-- Ubisoft disabled -->';
+    const URL = 'https://ubistatic2-a.akamaihd.net/worldwide_analytics/tagcommander/wikia/tc_WikiaEMEA_1.js';
+
+    /**
+     * @param array $params
+     * @return null
+     */
+    public function getSetupHtml( $params = array() ) {
+        return null;
+    }
+
+    /**
+     * @param $eventName
+     * @param array $eventDetails
+     * @return string
+     */
+    public function trackEvent( $eventName, $eventDetails = array() ) {
+        return self::isEnabled() ? $this->getScript() : self::DISABLED_MESSAGE;
+    }
+
+    /**
+     * @return bool
+     */
+    public static function isEnabled() {
+        global $wgEnableUbisoft, $wgNoExternals;
+
+        return $wgEnableUbisoft && !$wgNoExternals;
+    }
+
+    /**
+     * @return string
+     */
+    private function getScript() {
+        return \MustacheService::getInstance()->render( self::TEMPLATE_PATH, [
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)